std/string.d

```diff
@@ -355,12 +397,75 @@ ptrdiff_t indexOf(Char)(in Char[] s, in dchar c,
             }
         }
 
-        // c is a universal character
-        foreach (ptrdiff_t i, dchar c2; s)
+        static if (Char.sizeof == 1)
+        {
+            if (c <= 0x7F)
+            {
+                ptrdiff_t i;
+                foreach (const c2; s)
+                {
+                    if (c == c2)
+                        return i;
+                    ++i;
+                }
+            }
+            else
+            {
+                ptrdiff_t i;
+                foreach (const c2; s.byDchar())
+                {
+                    if (c == c2)
+                        return i;
+                    i += numberCodeUnits(c2);
+                }
```

Comment:
Usually the fastest loop is `while (i < s.length) if (decode(s, i) == c) return i;`, because it avoids the `codeLength` part, but it is only useable with random index strings. Maybe we can improve `byDchar` to provide optional iteration with index/counter.
Anyhow, you replaced the druntime based foreach decoding, so this is going to be faster.